packages/api/src/functions/secureHandler.ts

```diff
@@ -0,0 +1,214 @@
+/*
+
+Based on Stripe's secure webhook implementation
+See: https://stripe.com/docs/webhooks/signatures
+
+*/
+
+import { createHmac } from 'crypto'
+
+import { ForbiddenError } from 'apollo-server-lambda'
+import type { APIGatewayProxyEvent } from 'aws-lambda'
+
+/**
+ * @const {string}
+ */
+const ERROR_MESSAGE = "You don't have access to invoke this function."
+
+/**
+ * @const {number}
+ */
+const FIVE_MINUTES = 5 * 60 * 1000
+
+/**
+ * @const {number}
+ */
+const DEFAULT_TOLERANCE = FIVE_MINUTES
+
+/**
+ * @const {string}
+ */
+const WEBHOOK_SECRET = process.env['WEBHOOK_SECRET'] || ''
+
+/**
+ * @const {string}
+ */
+export const WEBHOOK_SIGNTAURE_HEADER = 'RW-WEBHOOK-SIGNATURE'
+
+/**
+ * VerifyOptions
+ * Used when verifying a signature's time component for permitted leeway.
+ *
+ * @typedef {Object} VerifyOptions
+ * @property {number} tolerance - Optional tolerance in msec
+ * @property {number} timestamp - Optional timestamp in msec
+ */
+export interface VerifyOptions {
+  tolerance?: number
+  timestamp?: number
+}
+
+/**
+ * Generates a hash-based message authentication code from a secret.
+ *
+ * @param {string} secret - Secret key used to sign
+ * @return {string} - A HMAC
+ *
+ * @example
+ *
+ *    getHmac({ secret: 'MY_SECRET' })
+ */
+const getHmac = ({ secret }: { secret: string }) => {
+  if (secret === undefined || secret === '')
```

Comment:
```suggestion
  if (typeof secret === 'undefined' || secret === '')
```
The check here is fine, since we know "secret" is defined, but I usually almost always stick to this for rather than `=== undefined`:

https://www.geeksforgeeks.org/variable-undefined-vs-typeof-variable-undefined-in-javascript/

Reply:
Done!